src/Nethermind/Nethermind.Evm/Instructions/EvmInstructions.Storage.cs

```diff
@@ -113,17 +114,19 @@ public static EvmExceptionType InstructionTStore(VirtualMachine vm, ref EvmStack
         StorageCell storageCell = new(vmState.Env.ExecutingAccount, in result);
 
         // Pop the 32-byte value from the stack.
-        Span<byte> bytes = stack.PopWord256();
+        ref var v = ref stack.PopBytesByRef();
+        if (Unsafe.IsNullRef(ref v)) return EvmExceptionType.StackUnderflow;
+
+        StorageValue value = new StorageValue(in Unsafe.As<byte, Vector256<byte>>(ref v));
```

Comment:
NP: Maybe add a constructor function `StorageValue.fromByteRef(ref v)`. This thing gets hard to follow real quick.

Reply:
Agree with the ugliness. A method extracted and moved to the stack. Also, there was the unneeded return here which was bloating a bit. Fixed this as well.